ngram.py

```diff
@@ -23,6 +21,9 @@ def __repr__(self):
     def __len__(self):
         return len(self.string)
 
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
```

Comment:
This is a quick and dirty way to compare class instances (http://stackoverflow.com/questions/1227121/compare-object-instances-for-equality-by-their-attributes-in-python). I don't like that it only exists for testing purposes but we can revisit this once we have more tests.